src-ts/tools/learn/learn-lib/my-certifications-provider/my-certification-progress.provider.tsx

```diff
@@ -1,16 +1,19 @@
 import { Dispatch, SetStateAction, useCallback, useEffect, useMemo, useState } from 'react'
 
-import { LearnMyCertification } from './learn-my-certification.model'
 import { MyCertificationProgressProviderData } from './my-certification-progress-provider-data.model'
-import myCertificationsJSON from './my-certifications-functions/my-certifications.json'
+import { getMyCertificationsProgressAsync, LearnMyCertificationProgress } from './my-certifications-functions'
+import { decorateCompletedPercentage, mapCompletedPercentage } from './my-certifications-functions/certificate-progress.decorators'
 
-const myCertifications: Array<LearnMyCertification> = [...myCertificationsJSON.certifications] as Array<LearnMyCertification>
+export function useMyCertificationProgress(userId?: number, certification?: string): MyCertificationProgressProviderData {
+    function setCertificateProgress(progress: LearnMyCertificationProgress): void {
+        setState((prevState) => ({...prevState, certificateProgress: decorateCompletedPercentage(progress)}))
```

Comment:
Can we instead get the completed percentage back in the updated API call?

Reply:
Hey @testflyjets ! I'm not sure I understand this one. Do you want me to send the certificateProgress percentage back to the PUT API call?

Reply:
Does the `decorateCompletedPercentage` method add the computed course completion? From my previous comment I thought you were computing the module completion but if this is course completion, this is fine until I am able to add it to the API response.